lib/ansible/modules/source_control/git.py

```diff
@@ -977,6 +988,57 @@ def main():
     track_submodules = module.params['track_submodules']
 
     result.update(before=None)
+
+    if archive:
+        if module.check_mode:
+            result.update(changed=True)
+            module.exit_json(**result)
+
+        # Archive already exists, then fail
+        if os.path.exists(archive):
+            module.fail_json(msg="Archive file %s already exists" % archive,
```

Comment:
This is going to bite people if they run this module twice. Ideally, on second run through, the changed should result in False if the archive hasn't changed, although I'm not sure of a particularly feasible way to do this (other than to generate the new archive in a temporary directory and compare the hashes of the archives)